### PR TITLE
fix: fix/slim up Strapi Dockerfile

### PR DESCRIPTION
**Pull request by vidriduch**
* Almost entirely change Dockerfile 
  - Switch to alpine distribution based on (un)official docker recommendations https://docs.strapi.io/developer-docs/latest/setup-deployment-guides/installation/docker.html#production-environments
  - Separate Dockerfile into multiple stages to keep DRY as possible, to always make change only in one place
  - Added tini as entrypoint to the image
  - Added opencontainers labels for the image and ENV variable which should hold current commit hash (if build with --build-args)
  - Limit the number of files that are copied to the final image. Kept only the necessary files to start up strapi (this might need some changes in the future)
  - Added comments for future use/copy to other projects about plugins and patches
  - Keep development stage without build as you probably want to mount your own file
* Move one package from plugin to main package.json, which helped us to limit the size of image by 400MB
* Added more stuff into .dockerignore as in the current form it copied a lot of unnecessary files, which slowed down the build time

### CODE REVIEW

File: strapi/Dockerfile [hunk 1]
```diff
@@ -1,57 +1,66 @@
...
-COPY yarn.lock ./
+FROM node:16-alpine AS build-base
+RUN apk update \
+ && apk add --no-cache build-base \
+                       gcc \
+                       autoconf \
+                       automake \
+                       zlib-dev \
+                       libpng-dev \
+                       vips-dev > /dev/null 2>&1
```
Comment (VeeeneX) [inline strapi/Dockerfile:9]: Why you need this `> /dev/null 2>&1`?

File: strapi/Dockerfile [hunk 2]
```diff
@@ -1,57 +1,66 @@
...
+FROM install-prod as build
+WORKDIR /build/app
+COPY ./ .
+RUN yarn install --development && yarn build
 
+FROM node:16-alpine AS app-base
+RUN apk add --no-cache vips-dev \
+                       tini \
+ && mkdir -p /home/node/app \
+ && chown node:node /home/node/app
```
Comment (VeeeneX) [inline strapi/Dockerfile:33]: I would split these into apk add and then mkdir with chown

File: strapi/package.json [hunk 3]
```diff
@@ -27,7 +27,8 @@
     "strapi-plugin-meilisearch": "^0.9.1",
     "strapi-plugin-navigation": "^2.2.1",
     "strapi-plugin-placeholder": "^4.3.6",
-    "strapi-provider-upload-ts-minio": "^3.1.0"
+    "strapi-provider-upload-ts-minio": "^3.1.0",
+    "xlsx": "^0.18.5"
```
Comment (VeeeneX) [inline strapi/package.json:31]: Did you updated or added xlsx?

File: strapi/Dockerfile (hunk 1 above)
Comment (vidriduch) [reply strapi/Dockerfile]: yes, this was a mistake and was left over as I copied it from the (un)official strapi documentation of package needed to build

File: strapi/Dockerfile (hunk 2 above)
Comment (vidriduch) [reply strapi/Dockerfile]: That is interesting notion. Adding one more layer so more layers can be just reused and when packaged to hub keeping it smaller. But on the other hand still adding one more layer, but as it didn't increase change much about the image I split the command into two RUN instructions as it probably makes more sense that way.

File: strapi/package.json (hunk 3 above)
Comment (vidriduch) [reply strapi/package.json:31]: I added xlsx as it is needed to run the strapi plugin that is part of this repository.

**Pull request merged**